crt_portal/static/js/constant_writer.js

```diff
@@ -0,0 +1,33 @@
+(function(root, dom) {
+  function getCWs(targets, reports) {
+    for (const [id, report] of Object.entries(reports.reports)) {
+      if (!report.constant_writer) continue;
+      const target = Array.from(targets).find(target => target.getAttribute('data-id') === id);
+      target.classList.remove('hidden');
+    }
+  }
+
+  function makeQuery() {
+    const targets = dom.querySelectorAll('.show-cw');
+    const report_objects = {};
+    targets.forEach(target => {
+      report_objects[target.dataset.id] = target.getAttribute('data-email');
```

Comment:
sometimes data-email is blank:

```
data-email="{{ datum.report.contact_email|default:'' }}">
```

does this code handle that properly? Would an `if (!getAttribute(data-email)) return)` make sense here?